src/components/recommendation.tsx

```diff
@@ -48,37 +50,104 @@ export function Recommendation({
   updateChat: (uuid: string, _chat: Partial<Chat>) => void;
 }) {
   const { isLoading, error } = useRecommendationQuery(chat, updateChat);
+  const [loadingMessage, setLoadingMessage] =
+    useState<string>(getRandomMessage());
 
-  if (isLoading) {
-    //TODO Loading UI
-    return <div>Loading...</div>;
-  }
-
-  if (error) {
-    //TODO Error
-    return <div>Error loading recommendations</div>;
-  }
+  useEffect(() => {
+    const intervalDuration =
+      (loadingMessage.split(" ").length / (100 / 60)) * 1000; //average reader WPM - 238
+    const interval = setInterval(() => {
+      setLoadingMessage(getRandomMessage());
+    }, intervalDuration);
+    return () => {
+      clearInterval(interval);
+    };
+  }, [loadingMessage]);
 
   return (
     <div className="w-full max-w-7xl space-y-4 p-8">
-      <div className="flex items-center gap-4">
-        <PromochatorIcon
-          imageWidth={40}
-          imageHeight={40}
-          imageClassName="py-2 px-1"
-        />
-        <p className="rounded-2xl bg-message-primary px-4 py-3">
-          {chat.helloMessage}
-        </p>
-      </div>
-      {chat.supervisors?.map((supervisor) => (
-        <Supervisor
-          key={supervisor.uuid}
-          supervisor={supervisor}
-          prompt={chat.prompt}
-          chatUuid={chat.uuid}
-        />
-      ))}
+      {isLoading ? (
+        <div className="flex items-center gap-x-4">
+          <PromochatorIcon
+            imageWidth={40}
+            imageHeight={40}
+            imageClassName="py-2 px-1"
+          />
+
+          <LoaderPinwheel
+            className="animate-spin"
+            style={{
+              animation: "spin 3s linear infinite",
+            }}
+            size={24}
+          />
+          <span className="w-full">{loadingMessage}</span>
+        </div>
+      ) : error ? (
+        <div className="flex gap-x-4 text-red-300">
+          <CircleX size={32}></CircleX>
+          <span className="w-4/5">
+            Wystąpił błąd przy pobieraniu rekomendacji. <br></br>Spróbuj
```

Comment:
**issue:** Line break is a self closing tag just like `<img>`
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/br
```suggestion
            Wystąpił błąd przy pobieraniu rekomendacji. <br/> Spróbuj
```